dadjokes-onchain/next/src/hooks/useWallet.js

```diff
@@ -0,0 +1,28 @@
+import { useState } from "react";
+import { ConnectWalletClient } from "@/lib/client";
+import { formatEther } from "viem";
+
+export function useWallet(dadJokesContract) {
+  const [address, setAddress] = useState(null);
+  const [balance, setBalance] = useState(null);
+
+  async function handleClick() {
+    try {
+      const walletClient = await ConnectWalletClient();
+      if (walletClient !== null) {
+        const [address] = await walletClient.requestAddresses();
+        setAddress(address);
+        // Retrieve the balance of the address using the Public Client
+        const balance = parseInt(
+          await dadJokesContract.read.creatorBalances([address])
```

Comment:
should the reading of balances be separated from the useWallet "concern"? feels like a non-intuitive place to find this piece of code?

Reply:
This checks to see if you have a balance you can withdraw when you connect your wallet. It seemed like the best place to implement that logic. It happens all at once.